src/aws/lambda-mediapackage/src/utils/mergeRecordSlicesManifest/index.js

```diff
@@ -0,0 +1,140 @@
+/**
+ * Post an update to the state of an object to any endpoint, using hmac with a shared secret
+ * to sign the object key and provide authorization for the request.
+ */
+
+const AWS = require('aws-sdk');
+const fetch = require('node-fetch');
+const { Parser } = require('m3u8-parser');
+
+const { CLOUDFRONT_ENDPOINT, DESTINATION_BUCKET_NAME } = process.env;
+
+const s3 = new AWS.S3({ apiVersion: '2006-03-01' });
+
+module.exports = async (environment, pk, stamp, recordSlices = []) => {
+  const mainManifestKey = `${pk}/cmaf/${stamp}.m3u8`;
+
+  const mainManifestParser = new Parser();
+  let mainManifestsLines = [];
+  let subManifests = {};
+
+  for (const recording_slice of recordSlices) {
+    // fetch the manifest content
+    const manifestUrl = `https://${CLOUDFRONT_ENDPOINT}/${recording_slice.manifest_key}`;
+    const response = await fetch(manifestUrl);
+    const manifest = await response.text();
+
+    console.log('Recording slice:', recording_slice);
+    console.log('Recording slice manifest url:', manifestUrl);
+    console.log('Recording slice manifest content:', manifest);
+
+    // parse the manifest
+    const hlsParser = new Parser();
+    hlsParser.push(manifest);
+    hlsParser.end();
+
+    if (mainManifestParser.lineStream.buffer.length === 0) {
+      mainManifestParser.push(manifest);
+      mainManifestParser.end();
+    }
+
+    await Promise.all(
+      hlsParser.manifest.playlists.map(async (playlist) => {
+        // collect main manifest lines
+        const resolution = playlist.attributes.RESOLUTION;
+        if (!mainManifestsLines.join().includes(resolution.height)) {
+          let attributes = `BANDWIDTH=${playlist.attributes.BANDWIDTH},`;
+          attributes += `AVERAGE-BANDWIDTH=${playlist.attributes['AVERAGE-BANDWIDTH']},`;
+          attributes += `RESOLUTION=${resolution.width}x${resolution.height},`;
+          attributes += `FRAME-RATE=${playlist.attributes['FRAME-RATE']},`;
+          attributes += `CODECS="${playlist.attributes['CODECS']}"`;
+
+          mainManifestsLines.push(`#EXT-X-STREAM-INF:${attributes}
+${environment}_${pk}_${stamp}_hls_${resolution.height}.m3u8`);
+        }
+
+        // fetch the sub manifest content
+        const manifestUrl = `https://${CLOUDFRONT_ENDPOINT}/${pk}/cmaf/${recording_slice.harvested_directory}/${playlist.uri}`;
+        const response = await fetch(manifestUrl);
+        const manifest = await response.text();
```

Comment:
renaming it in `subManifest` should ease to read this part of the code. We code confuse it with the one initialized line 25

Reply:
I renamed it with `harvestedSubManifest`, same with `harvestedMainManifest` for line 25. WDYT ?